crates/biome_js_analyze/src/lint/nursery/use_import_extensions.rs

```diff
@@ -82,18 +107,40 @@ declare_rule! {
     }
 }
 
+#[derive(Clone, Debug, Default, Deserializable, Deserialize, Serialize, Eq, PartialEq)]
+#[cfg_attr(feature = "schemars", derive(JsonSchema))]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+pub struct UseImportExtensionsOptions {
+    /// List of custom import extension mappings
+    pub import_mappings: Vec<ImportExtensionMapping>,
+}
+
+#[derive(Debug, Clone, Default, Deserializable, Deserialize, Serialize, Eq, PartialEq)]
+#[cfg_attr(feature = "schema", derive(schemars::JsonSchema))]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+pub struct ImportExtensionMapping {
+    /// List of inspected file extensions that are subject to this mapping
+    pub file_ext: Vec<String>,
+    /// Extension that should be used for module imports
+    pub import_ext: String,
```

Comment:
What do you mean? Isn't the whole idea behing this PR to fix the auto-fix functionality, which currently uses the extension that is wrong for Node.js environment?

Reply:
This rule only checks for imports without extensions. Biome does not have infra yet, to check for filesystem. This PR won't fix wrong extensions. It will only suggested user selected extension when there isn't one.

I can see checking for wrong extensions added in a future, so my naming change can be ignored, though I imagine that won't be easy to get right in all cases.

Reply:
OK, so to clarify: `biome check --fix --unsafe` currently does add missing extension when there isn't one at all. That is expected, and after this change the user-specified extension will be applied. Is that correct?

What you are saying is that if e. g. original file is "file.ts", and the import in code is "file.rtf", then Biome will say that it's perfectly fine, right?

Reply:
Yes. `import { named } from './file.rtf'` is valid import as far as this rule sees it.